jcl/src/java.base/share/classes/com/ibm/tools/attach/target/CommonDirectory.java

```diff
@@ -126,32 +128,64 @@ static void prepareCommonDirectory() throws IOException {
 	 * Lock the master lock file. Create the lockfile if necessary.
 	 * @throws IOException if the file is already locked.
 	 */
-	/*[PR Jazz 30075] obtainMasterLock(File commonDirectory) is never used. */		
 	public static void obtainMasterLock() throws IOException {
-		getMasterLock().lockFile(true);
+		FileLock masterLockCopy = null;
+		synchronized (accessorMutex) {
+			++masterLockCount;
+			if (1 == masterLockCount) {
+				masterLockCopy = getMasterLock();
+			}
+		}
+		if (null != masterLockCopy) { /* first entry */
+			masterLockCopy.lockFile(true);
+		}
 	}
 
 	/**
-	 * non-blocking lock the master lockfile. Create the lockfile if necessary.
+	 * non-blocking lock the master lockfile. 
+	 * Create the lockfile if necessary.
 	 * @return true if lock obtained
-	 * @throws IOException if file already locked
 	 */
-	static boolean tryObtainMasterLock() throws IOException {
-		return getMasterLock().lockFile(false);
+	static boolean tryObtainMasterLock() {
+		boolean masterLockEntered = true;
+		synchronized (accessorMutex) {
+			++masterLockCount; /* optimistically assume we enter the lock */
+			if (1 == masterLockCount) { /* first time in */
+				try {
+					masterLockEntered = getMasterLock().lockFile(false);
+					if (!masterLockEntered) { /* lock failed, so revert */
+						--masterLockCount;
+					}
+				} catch (IOException e) { /* this shouldn't happen */
+					masterLockEntered = false;
+					IPC.logMessage("IOException in tryObtainMasterLock, masterLockCount="+masterLockCount, //$NON-NLS-1$
+							e);
```

Comment:
Seems weird to have such a short continuation, it could be included on the previous line.